examples/hpc/dla-driver/src/utils.rs

```diff
@@ -0,0 +1,123 @@
+#![no_std]
+
+use crate::mmap::MEMORY_BANK_SIZE;
+use crate::tensor3::{Order3, Tensor3};
+use crate::tensor4::Tensor4;
+use crate::{MemoryBank, Padding, Stride, DEFAULT_PADDING, DEFAULT_STRIDE, MEMORY_BANK_BASE_ADDR};
+use alloc::vec::Vec;
+
+/// Calculates the output size of Conv2D for a single channel based on size of the inputs
+///
+/// * `input` - Input data for a given layer.
+/// * `kernel` - Kernels/weight data for a given layer.
+/// * `padding` - Padding used in the given layer.
+/// * `stride` - Stride used in the given layer.
+pub fn calculate_conv2d_out_param_dim(
+    input: (u32, u32),
+    kernel: (u32, u32),
+    padding: Option<Padding>,
+    stride: Option<Stride>,
+) -> (usize, usize) {
+    let padding = padding.unwrap_or(DEFAULT_PADDING);
+    let stride = stride.unwrap_or(DEFAULT_STRIDE);
+
+    let output_width =
+        (input.0 + padding.right + padding.left - 1 * (kernel.0 - 1) - 1) / stride.x + 1;
+    let output_height =
+        (input.1 + padding.bottom + padding.top - 1 * (kernel.1 - 1) - 1) / stride.y + 1;
+    (output_width as usize, output_height as usize)
+}
+
+/// Creates a output tensor matching the given inputs from the ground truth output.
+///
+/// * `input` - Input data for a given layer.
+/// * `kernel` - Kernels/weight data for a given layer.
+/// * `output_buf` - Data containing the known ground truth for the output of a layer.
+/// * `order` - Order of the data in output_buf.
+/// * `padding` - Padding used in the given layer.
+/// * `stride` - Stride used in the given layer.
+pub fn generate_output_tensor<T: Clone, H: Clone, J: Clone>(
+    input: &Tensor3<T>,
+    kernel: &Tensor4<H>,
+    output_buf: Vec<J>,
+    order: Order3,
+    padding: Option<Padding>,
+    stride: Option<Stride>,
+) -> Tensor3<J> {
```

Comment:
Generic names `I, K, O` might be more memorizable.

Reply:
Fixed